tool/tsh/tsh.go

```diff
@@ -3391,6 +3391,19 @@ func makeClientForProxy(cf *CLIConf, proxy string) (*client.TeleportClient, erro
 		}
 	}
 
+	// When using Headless, check for missing proxy/user/cluster values from the teleport session env variables.
+	if cf.AuthConnector == constants.HeadlessConnector {
+		if c.WebProxyAddr == "" {
+			c.WebProxyAddr = os.Getenv(teleport.SSHSessionWebproxyAddr)
+		}
+		if c.Username == "" {
+			c.Username = os.Getenv(teleport.SSHTeleportUser)
+		}
```

Comment:
how does this interact with the ExplicitUsername check above? Should we also be doing the ExplicitUsername check when the user provides `--auth=headless` instead of `--headless`?

Reply:
Yes, and it should also count `SSHTeleportUser` as explicity username. Thanks for catching this.